File: 03330. Find the Original Typed String I/solution.js
/**
 * Problem: 3330. Find the Original Typed String I
 *
 * Difficulty: Easy
 *
 * Language: JavaScript
 *
 * Performance: Runtime - 28 ms (Beats 100%)
 */

/**
 * Calculates the number of possible strings by removing consecutive duplicate characters
 *
 * @param {string} word - Input string with consecutive identical characters
 *
 * @returns {number} - Number of possible strings
 */
const possibleStringCount = (word) =>
  word.match(/(.)\1*/g).reduce((sum, group) => sum + group.length - 1, 1)

File: 03330. Find the Original Typed String I/solution.go
/**
 * Problem: 3330. Find the Original Typed String I
 *
 * Difficulty: Easy
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func possibleStringCount(word string) int {
	// Get the length of the input word and initialize the count to 1
	wordLength, count := len(word), 1
	
	// Iterate through the word to count consecutive duplicate characters
	for i := 1; i < wordLength; i++ {
		// If current character matches the previous one, increment count
		if word[i-1] == word[i] {
			count++
		}
	}

	// Return the total count of possible original strings
	return count
}
